components/hab/src/main.rs

```diff
@@ -822,6 +822,14 @@ fn sub_pkg_upload(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     // before allowing a write to the backend, this bypasses the check
     let force_upload = m.is_present("FORCE");
 
+    // Builder checks for presence of the 'builder' param to determine
+    // if a job group should be created. 
+    let disable_build_group = if m.is_present("NO_BUILD") {
```

Comment:
It might be simpler/less code to just pass and use this as a bool like `force_upload`